shared/src/main/scala/implicits.scala

```diff
@@ -16,29 +16,69 @@
 
 package fetch
 
-import cats.MonadError
+import java.util.{Timer, TimerTask}
+import java.util.concurrent.TimeoutException
+import scala.concurrent.duration._
 import cats.instances.future._
 import scala.concurrent.{ExecutionContext, Future, Promise}
 
 object implicits {
 
-  implicit def fetchFutureFetchMonadError(
-      implicit ec: ExecutionContext
-  ): FetchMonadError[Future] =
+  // Shared Timer object to schedule timeouts
+  private[fetch] lazy val timer: Timer = new Timer(true)
```

Comment:
yeah good idea, will help avoid confusion when people look at running threads and don't know what it's for